provider/cldr/src/cldr_serde/week_data.rs

```diff
@@ -0,0 +1,123 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Serde structs representing CLDR JSON weekData.json files.
+//!
+//! Sample file:
+//! https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/weekData.json
+
+use core::convert::TryFrom;
+use serde::{Deserialize, Deserializer};
+use std::collections::BTreeMap;
+use std::num::ParseIntError;
+use std::str::FromStr;
+use tinystr::{tinystr4, TinyStr4};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum Weekday {
+    Mon,
+    Tue,
+    Wed,
+    Thu,
+    Fri,
+    Sat,
+    Sun,
+}
+
+impl From<&Weekday> for icu_calendar::types::IsoWeekday {
+    fn from(day: &Weekday) -> Self {
+        use icu_calendar::types::IsoWeekday;
+        match day {
+            Weekday::Mon => IsoWeekday::Monday,
+            Weekday::Tue => IsoWeekday::Tuesday,
+            Weekday::Wed => IsoWeekday::Wednesday,
+            Weekday::Thu => IsoWeekday::Thursday,
+            Weekday::Fri => IsoWeekday::Friday,
+            Weekday::Sat => IsoWeekday::Saturday,
+            Weekday::Sun => IsoWeekday::Sunday,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Territory {
```

Comment:
suggestion: This could use a small bit of code docs. I wasn't able to map it to the .json data linked above easily. It would be nice if this explained the justification for using `TinyStr4`.

Reply:
Done. The choice of TinyStr4 was to avoid an extra parsing step in the data provider (variant -> region -> map lookup)